lib/plug/conn/query.ex

```diff
@@ -186,8 +186,12 @@ defmodule Plug.Conn.Query do
 
   # covers non-keyword lists
   defp encode_pair(parent_field, list, encoder) when is_list(list) do
-    prune Enum.flat_map list, fn value ->
-      [?&, encode_pair(parent_field <> "[]", value, encoder)]
+    prune Enum.flat_map list, fn
+      value when is_map(value) and map_size(value) != 1 ->
+        raise ArgumentError,
+          "cannot encode maps with 0 or more than 1 elements, got: #{inspect(value)}"
```

Comment:
Sure, that sounds a bit clearer. Updated :)

Btw, do you think `ArgumentError` is fine? It could be `Plug.Conn.InvalidQueryError` but I'm not sure 😕